modules/programs/anki.nix

```diff
@@ -0,0 +1,567 @@
+        type = with lib.types; nullOr str;
+        default = null;
+        example = "http://example.com/anki-sync/";
+        description = ''
+          Custom sync server URL. See <https://docs.ankiweb.net/sync-server.html>.
+        '';
+      };
+
+      autoSync = lib.mkOption {
+        type = with lib.types; nullOr bool;
+        default = null;
+        example = true;
+        description = "Automatically sync on profile open/close.";
+      };
+
+      syncMedia = lib.mkOption {
+        type = with lib.types; nullOr bool;
+        default = null;
+        example = true;
+        description = "Synchronize audio and images too.";
+      };
+
+      autoSyncMediaMinutes = lib.mkOption {
+        type = with lib.types; nullOr ints.unsigned;
+        default = null;
+        example = 15;
+        description = ''
+          Automatically sync media every X minutes. Set this to 0 to disable
+          periodic media syncing.
+        '';
+      };
+
+      networkTimeout = lib.mkOption {
+        type = with lib.types; nullOr ints.unsigned;
+        default = null;
+        example = 60;
+        description = "Network timeout in seconds.";
+      };
+    };
+
+    addons = lib.mkOption {
+      type = with lib.types; listOf package;
+      default = [ ];
+      example = lib.literalExpression ''
+        [
+          # When the add-on is already available in nixpkgs
+          pkgs.ankiAddons.anki-connect
+
+          # When the add-on is not available in nixpkgs
+          (pkgs.anki-utils.buildAnkiAddon (finalAttrs: {
+            pname = "recolor";
+            version = "3.1";
+            src = pkgs.fetchFromGitHub {
+              owner = "AnKing-VIP";
+              repo = "AnkiRecolor";
+              rev = finalAttrs.version;
+              sparseCheckout = [ "src/addon" ];
+              hash = "sha256-28DJq2l9DP8O6OsbNQCZ0pm4S6CQ3Yz0Vfvlj+iQw8Y=";
+            };
+            sourceRoot = "''${finalAttrs.src.name}/src/addon";
+          }))
+
+          # When the add-on needs to be configured
+          pkgs.ankiAddons.passfail2.withConfig {
+            config = {
+              again_button_name = "not quite";
+              good_button_name = "excellent";
+            };
+            user_files = ./dir-to-be-merged-into-addon-user-files-dir;
+          };
+        ]
+      '';
+      description = "List of Anki add-on packages to install.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.sync.username != null && cfg.sync.usernameFile != null);
+        message = ''
+          The `programs.anki.sync` `username` option is mutually exclusive with
+          the `usernameFile` option.
+        '';
+      }
+      {
+        assertion = cfg.package ? withAddons;
+        message = ''
+          The value of `programs.anki.package` doesn't support declaratively managing
+          add-ons. Make sure you are using `pkgs.anki`.
+        '';
+      }
```

Comment:
Why not support a nullable package and change the logic in `home.packages` to either

- Install no package
- Install package without addons
- Install package with addons